### PR TITLE
Move the config to being passed as a command line flag.

### PR DESCRIPTION
**Pull request by johndaniels**
We shouldn't have the config be a checked in, relatively imported
file. This makes it easy to check in tokens accidentally and makes
it hard to run the bot using npm install. This change adds
a command line flag '--config' to specify a JSON config file.

### CODE REVIEW

File: src/bot/config.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Gets the specified key from a JSON, raising an exception if the value is null/undefined
+ */
+function getNonNull(jsonObject: any, key: string) {
+    if (jsonObject[key] == null) {
+        throw new Error('Missing required configuration key: ' + key);
+    }
+    return jsonObject[key];
+}
```
Comment (kata-gatame) [inline src/bot/config.ts]: I feel like this would best fit in `src/lib/util/index`.

File: src/bot/config.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        this.token = getNonNull(jsonObject, 'token');
+        this.owner = getNonNull(jsonObject, 'owner');
+        this.commandPrefix = getNonNull(jsonObject, 'commandPrefix');
+        this.statusType = getNonNull(jsonObject, 'statusType');
+        this.statusText = getNonNull(jsonObject, 'statusText');
+    }
+}
+
+const rawConfig = fs.readFileSync(yargs.argv.config as string, 'utf8');
+export default new Configuration(JSON.parse(rawConfig));
```
Comment (kata-gatame) [inline src/bot/config.ts]: I think the best spot for this would be `src/lib/classes/`.

Comment (johndaniels) [reply src/bot/config.ts]: I'm not 100% on what the dividing line is for the lib folder vs bot. Should I just put the Configuration class in `src/lib/classes` and leave the initialization here?

**Comment by johndaniels** [comment]
I didn't have VSCode set up right. Now it's giving me errors properly. Also running the linter by hand to be sure. Everything should be fixed.

File: src/bot/config.ts (hunk 2 above)
Comment (kata-gatame) [reply src/bot/config.ts]: That's a good point to bring up.  I've been bouncing back and forth on structure.

There are classes that are under bot folder, which could go under the lib directory as well.  I think we should have a quick discord conversation before moving forward much further about our structure.

**Pull request merged**